src/BudgetManager/BudgetManager.CQRS/Handlers/WalletHandlers/GetUserWalletTransactionsHandler.cs

```diff
@@ -0,0 +1,40 @@
+﻿using BudgetManager.CQRS.Queries.WalletQueries;
+using BudgetManager.Model;
+using BudgetManager.SDK.DTOs;
+using BudgetManager.Shared.DataAccess.MongoDB.BaseImplementation;
+using MediatR;
+using MongoDB.Driver;
+
+namespace BudgetManager.CQRS.Handlers.WalletHandlers
+{
+    public class GetUserWalletTransactionsHandler : IRequestHandler<GetUserWalletTransactionsQuery, WalletTransactionsDTO>
+    {
+        private readonly IBaseRepository<User> _userRepository;
+        private readonly IBaseRepository<Transaction> _transactionRepository;
+
+        public GetUserWalletTransactionsHandler(IBaseRepository<User> userRepository, IBaseRepository<Transaction> transactionRepository)
+        {
+            _userRepository = userRepository;
+            _transactionRepository = transactionRepository;
+        }
+
+        public async Task<WalletTransactionsDTO> Handle(GetUserWalletTransactionsQuery request, CancellationToken cancellationToken)
+        {
+            User user = await _userRepository.FindByIdAsync(request.userId, cancellationToken);
+            Wallet wallet = user.Wallets.Find(w => w.Id == request.walletId);
+
+            var builder = Builders<Transaction>.Filter;
+            var transactionFilter = builder.Eq(t => t.UserId, request.userId) & builder.Eq(t => t.WalletId, request.walletId);
+            var userWalletTransactions = await _transactionRepository.FilterBy(transactionFilter, cancellationToken);
+
+            WalletTransactionsDTO walletTransactions = new WalletTransactionsDTO
+            {
+                Wallet = wallet,
+                IsDefaultWallet = wallet.Id == request.walletId,
+                Transactions = userWalletTransactions.ToList(),
+            };
```

Comment:
It is not good option to have Filtering methods in Handlers. I have created ITransactionRepository interface you can apply this function in TransactionRepo and use it here

Reply:
And this function should have SortBy(x=>x.TransactionDate) to give user sorted transaction so that he can see from the latest to oldest

Reply:
Ah, yes, sorry. I forgot.
Probably, it should be OrderByDescending(t => t.TransactionDate), because the transaction date should be from the most recent to the oldest ones.

Reply:
> 

I think it is not a problem to filter in handler. Handlers are created for business logic and repositories only for retrieving the data from DB. In our case filtration is based on business requirements that's why it is a business logic

Reply:
> It is not good option to have Filtering methods in Handlers. I have created ITransactionRepository interface you can apply this function in TransactionRepo and use it here

I think there's no need for that, since our generic base repository method accepts filter conditions for data retrieval